source/globalCommands.py

```diff
@@ -569,6 +569,24 @@ def script_previousSynthSetting(self, gesture):
 		previousSettingValue = globalVars.settingsRing.currentSettingValue
 		ui.message("%s %s" % (previousSettingName, previousSettingValue))
 
+	@script(
+		# Translators: Input help mode message for toggling keyboard layout.
+		description=_("Toggles keyboard layout (desktop or laptop)"),
+		category=SCRCAT_INPUT,
+	)
+	def script_toggleKeyboardLayout(self, gesture: "inputCore.InputGesture") -> None:
+		val = config.conf["keyboard"]["keyboardLayout"]
+		if val == "desktop":
+			newVal = "laptop"
+		else:
+			newVal = "desktop"
+		config.conf["keyboard"]["keyboardLayout"] = newVal
+		# Translators: Reported when the user toggles keyboard layout.
+		# {layout} will be replaced with the new keyboard layout, i.e. desktop or laptop
+		ui.message(
+			_("Keyboard layout: {layout}").format(layout=keyboardHandler.KeyboardInputGesture.LAYOUTS[newVal])
```

Comment:
I think the colon is unnecessary here.
```suggestion
			_("Keyboard layout {layout}").format(layout=keyboardHandler.KeyboardInputGesture.LAYOUTS[newVal])
```
We could potentially also get away without saying "keyboard" for brevity
```suggestion
			_("Layout {layout}").format(layout=keyboardHandler.KeyboardInputGesture.LAYOUTS[newVal])
```
Or, to put the change first
```suggestion
			_("{layout} layout").format(layout=keyboardHandler.KeyboardInputGesture.LAYOUTS[newVal])
```
What do you think?